### PR TITLE
adjust time refactoring

### PR DESCRIPTION
**Pull request by zargham-ahmad**
This include the first refactoring of the adjust time function for #36

### CODE REVIEW

File: R/adjust.time.R [hunk 1]
```diff
@@ -1,3 +1,78 @@
...
+                           x.points = to.correct)
+    
+    this.corrected[this.old >= min(this.dev) & this.old <= max(this.dev)] <- 
+      this.smooth$y + to.correct
+    this.corrected[this.old < min(this.dev)] <- 
+      this.corrected[this.old < min(this.dev)] + 
+      mean(this.smooth$y[this.smooth$x == min(this.smooth$x)])
+    this.corrected[this.old > max(this.dev)] <- 
+      this.corrected[this.old > max(this.dev)] + 
+      mean(this.smooth$y[this.smooth$x == max(this.smooth$x)])
```
Comment (xtrojak) [inline R/adjust.time.R]: A little bit hard to read.

File: R/adjust.time.R [hunk 2]
```diff
@@ -12,9 +87,14 @@ adjust.time <- function(features,
...
-                 ylab="", main="", axes=FALSE)
-            text(x=0, y=0, "Retention time \n adjustment", cex=2)
+            draw_plot(x = c(-1,1), 
+                      y = c(-1,1), 
+                      xlab = "", 
+                      ylab = "", 
+                      main = "", 
+                      axes = FALSE, 
+                      label = "Retention time \n adjustment", 
+                      cex = 2)
```
Comment (xtrojak) [inline R/adjust.time.R]: Do you need to pass default arguments?

File: R/adjust.time.R [hunk 3]
```diff
@@ -25,15 +105,25 @@ adjust.time <- function(features,
...
-                 ylab="", main="m/z tolerance level given", axes=FALSE)
-            text(x=0, y=0, mz.tol, cex=1.2)
+            draw_plot(x = c(-1,1), 
+                      y = c(-1,1), 
+                      xlab = "", 
+                      ylab = "", 
+                      main = "m/z tolerance level given", 
+                      axes = FALSE, 
+                      label = mz.tol, 
+                      cex = 1.2)
```
Comment (xtrojak) [inline R/adjust.time.R]: Same.

File: R/adjust.time.R [hunk 4]
```diff
@@ -25,15 +105,25 @@ adjust.time <- function(features,
...
-                 ylab="", main="retention time \n tolerance level given", axes=FALSE)
-            text(x=0, y=0, chr.tol, cex=1.2)
+            draw_plot(x = c(-1,1), 
+                      y = c(-1,1), 
+                      xlab = "", 
+                      ylab = "", 
+                      main = "retention time \n tolerance level given", 
+                      axes = FALSE, 
+                      label = chr.tol, 
+                      cex = 1.2)
```
Comment (xtrojak) [inline R/adjust.time.R]: And one more time.

**Review by xtrojak — CHANGES_REQUESTED** [review]
Thank you @zargham-ahmad, I like how many functions you managed to extract! Please look at the code style, I feel like it's not consistent (I personally use automatic code reformatting in `Rstudio`).

**Review by xtrojak — APPROVED** [review]

**Review by hechth — APPROVED** [review]
Good work! Just for the future - if you respond to review comments, just indicate where you made the changes regarding this comment in the conversation - it is otherwise hard to track what has been handled where and when!

File: R/adjust.time.R (hunk 4 above)
Comment (zargham-ahmad) [reply R/adjust.time.R]: resolved in [024c3be](https://github.com/RECETOX/recetox-aplcms/pull/64/commits/024c3be853f22d9a0ff5a7a5769194cf9e74eb24)

**Pull request merged**